### PR TITLE
Fix concretes entities in mapping files

### PR DESCRIPTION
**Pull request by skeleton**
| Q                 | A
| ----------------- | ---
| Specs             | N
| Behats            | N
| Blue CI           | :white_check_mark: 
| Changelog updated | N
| Review and 2 GTM  |  :white_check_mark: 

Fix mapping files to add interface instead of concrete entity (integrators cannot override mapping if there is a concrete entity)

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Resources/config/model/doctrine/Association.mongodb.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ Pim\Component\Catalog\Model\Association:
         groups:
             notSaved: true
             type: entities
-            targetEntity: Pim\Bundle\CatalogBundle\Entity\Group
+            targetEntity: Pim\Component\Catalog\Model\GroupInterface
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Resources/config/model/doctrine/Association.mongodb.yml]: can't work right now
seen IRL

File: src/Pim/Bundle/CatalogBundle/Resources/config/model/doctrine/Completeness.mongodb.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ Pim\Component\Catalog\Model\Completeness:
             type: int
         channel:
             type: entity
-            targetEntity: Pim\Bundle\CatalogBundle\Entity\Channel
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Resources/config/model/doctrine/Completeness.mongodb.yml]: can't work right now
seen IRL

File: src/Pim/Bundle/CatalogBundle/Resources/config/model/doctrine/Product.mongodb.yml [hunk 3]
```diff
@@ -47,18 +47,18 @@ Pim\Component\Catalog\Model\Product:
             type: date
         family:
             type: entity
-            targetEntity: Pim\Bundle\CatalogBundle\Entity\Family
+            targetEntity: Pim\Component\Catalog\Model\FamilyInterface
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Resources/config/model/doctrine/Product.mongodb.yml]: can't work right now
seen IRL

File: src/Pim/Bundle/CatalogBundle/Resources/config/model/doctrine/ProductValue.mongodb.yml [hunk 4]
```diff
@@ -23,7 +23,7 @@ Pim\Component\Catalog\Model\ProductValue:
             type: string
         attribute:
             type: entity
-            targetEntity: Pim\Bundle\CatalogBundle\Entity\Attribute
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Resources/config/model/doctrine/ProductValue.mongodb.yml]: can't work right now
seen IRL

**Comment by jjanvier** [comment]
GTM except comment

**Comment by skeleton** [comment]
map an interface in mongo db mapping files, even if the interface is stored in ORM, works now

File: src/Akeneo/Bundle/StorageUtilsBundle/EventListener/MongoDBODM/ResolveTargetEntityListener.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Akeneo\Bundle\StorageUtilsBundle\EventListener\MongoDBODM;
+
+use Doctrine\ODM\MongoDB\Event\LoadClassMetadataEventArgs;
+use Doctrine\ODM\MongoDB\Mapping\ClassMetadata;
+
+/**
+ * ResolveTargetEntityListener
```
Comment (fitn) [inline src/Akeneo/Bundle/StorageUtilsBundle/EventListener/MongoDBODM/ResolveTargetEntityListener.php]: to remove

File: src/Pim/Bundle/CatalogBundle/Entity/AssociationTypeTranslation.php [hunk 6]
```diff
@@ -11,28 +12,18 @@
  * @copyright 2013 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-class AssociationTypeTranslation extends AbstractTranslation
+class AssociationTypeTranslation extends AbstractTranslation implements AttributeTypeTranslationInterface
```
Comment (fitn) [inline src/Pim/Bundle/CatalogBundle/Entity/AssociationTypeTranslation.php]: Implements AttributeTypeTranslationInterface? Naming issue?

File: src/Pim/Bundle/CatalogBundle/Entity/GroupTranslation.php [hunk 7]
```diff
@@ -3,6 +3,7 @@
 namespace Pim\Bundle\CatalogBundle\Entity;
 
 use Akeneo\Component\Localization\Model\AbstractTranslation;
+use Pim\Component\Catalog\Model\GroupTranslationInterface;
 
 /**
  * Variant group translation
```
Comment (fitn) [inline src/Pim/Bundle/CatalogBundle/Entity/GroupTranslation.php]: Group in fact. Not in your PR

File: src/Pim/Bundle/CatalogBundle/Entity/AssociationTypeTranslation.php (hunk 6 above)
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Entity/AssociationTypeTranslation.php]: good catch :)

File: src/Pim/Component/Catalog/Model/AssociationTypeTranslationInterface.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface AssociationTypeTranslationInterface extends TranslationInterface
+{
+    /**
+     * Set label
+     *
+     * @param string $label
+     *
+     * @return CategoryTranslationInterface
```
Comment (fitn) [inline src/Pim/Component/Catalog/Model/AssociationTypeTranslationInterface.php]: nope

File: src/Pim/Component/Catalog/Model/AttributeGroupTranslationInterface.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface AttributeGroupTranslationInterface extends TranslationInterface
+{
+    /**
+     * Set label
+     *
+     * @param string $label
+     *
+     * @return CategoryTranslationInterface
```
Comment (fitn) [inline src/Pim/Component/Catalog/Model/AttributeGroupTranslationInterface.php]: Nope. You can inheritdoc btw ;)

File: src/Pim/Component/Catalog/Model/AttributeTranslationInterface.php [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface AttributeTranslationInterface extends TranslationInterface
+{
+    /**
+     * Set label
+     *
+     * @param string $label
+     *
+     * @return CategoryTranslationInterface
```
Comment (fitn) [inline src/Pim/Component/Catalog/Model/AttributeTranslationInterface.php]: same

File: src/Pim/Component/Catalog/Model/AttributeTypeTranslationInterface.php [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface AttributeTypeTranslationInterface extends TranslationInterface
+{
+    /**
+     * Set label
+     *
+     * @param string $label
+     *
+     * @return CategoryTranslationInterface
```
Comment (fitn) [inline src/Pim/Component/Catalog/Model/AttributeTypeTranslationInterface.php]: same

File: src/Pim/Component/Catalog/Model/CategoryTranslationInterface.php [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface CategoryTranslationInterface extends TranslationInterface
+{
+    /**
+     * Set label
+     *
+     * @param string $label
+     *
+     * @return CategoryTranslationInterface
```
Comment (fitn) [inline src/Pim/Component/Catalog/Model/CategoryTranslationInterface.php]: This one is good :D

File: src/Pim/Component/Catalog/Model/GroupTypeTranslationInterface.php [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface GroupTypeTranslationInterface extends TranslationInterface
+{
+    /**
+     * Set label
+     *
+     * @param string $label
+     *
+     * @return CategoryTranslationInterface
```
Comment (fitn) [inline src/Pim/Component/Catalog/Model/GroupTypeTranslationInterface.php]: nope

File: src/Pim/Component/Catalog/Model/GroupTranslationInterface.php [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface GroupTranslationInterface extends TranslationInterface
+{
+    /**
+     * Set label
+     *
+     * @param string $label
+     *
+     * @return CategoryTranslationInterface
```
Comment (fitn) [inline src/Pim/Component/Catalog/Model/GroupTranslationInterface.php]: nope

File: src/Akeneo/Component/Localization/Model/TranslationInterface.php [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+    public function getId();
+
+    /**
+     * Set locale
+     *
+     * @param string $locale
+     *
+     * @return TranslationInterface
+     */
+    public function setLocale($locale);
```
Comment (fitn) [inline src/Akeneo/Component/Localization/Model/TranslationInterface.php]: Is it a string or a Locale. If string, it should be localeCode. If not possible (hard rework, huge bc, etc.), I don't care.

Comment (fitn) [inline src/Akeneo/Component/Localization/Model/TranslationInterface.php]: And I'm probably the owner of this naming issue :innocent:

**Comment by fitn** [comment]
Great work. It is a big improvement for integrators ;)

File: src/Akeneo/Component/Localization/Model/TranslationInterface.php (hunk 15 above)
Comment (skeleton) [inline src/Akeneo/Component/Localization/Model/TranslationInterface.php]: it's a string, but I prefere to keep setLocale (BC break..)

**Comment by nidup** [comment]
Yeah!

**Pull request merged**